app/views/checkout/payment.scala.html

```diff
@@ -34,6 +34,7 @@
             <form class="form js-checkout-form" method="POST">
 
                 @* ===== Your details ===== *@
+                <a name="yourDetails"></a>
```

Comment:
Could just give the `<fieldset>` and `id`? Same effect, but one-less element. Whilst I think about it, we should also do this for the **edit** links--same process but in reverse, and `<a`> pointing to the *previous* container.